README.md

```diff
@@ -6,22 +6,70 @@ Independent Parallel Particle Layer (IPPL) is a performance portable C++ library
 
 All IPPL releases (< 3.2.0) are available under the BSD 3-clause license. Since version 3.2.0, this repository includes a modified version of the `variant` header by GNU, created to support compilation under CUDA 12.2 with GCC 12.3.0. This header file is available under the same terms as the [GNU Standard Library](https://github.com/gcc-mirror/gcc); note the GNU runtime library exception. As long as this file is not removed, IPPL is available under GNU GPL version 3.
 
-## Installing IPPL and its dependencies
+# Installing IPPL and its dependencies
 
 All the new developments of IPPL are merged into the `master` branch which can make it potentially unstable from time to time. So if you want a stable and more tested version
 please checkout the tagged branch correspodning to the latest release (e.g. `git checkout tags/IPPL-x.x.x`). Otherwise if you want the latest developments go with the master with the above caveat in mind.
 
-### Requirements
-The following libraries are required:
+## Requirements
 
+* [CMake](https://cmake.org/download/)
+* A C++ compilation toolchain (GPU-capable for GPU builds, e.g. [nvcc](https://developer.nvidia.com/cuda-downloads), [clang]() or [rocmcc](https://rocm.docs.amd.com/en/docs-5.0.2/reference/rocmcc/rocmcc.html))
 * MPI (GPU-aware if building for GPUs)
-* [Kokkos](https://github.com/kokkos) >= 4.1.00
-* [HeFFTe](https://github.com/icl-utk-edu/heffte) >= 2.2.0; only required if IPPL is built with FFTs enabled (`ENABLE_FFT=ON`)
 
-To build IPPL and its dependencies, we recommend using the [IPPL build scripts](https://github.com/IPPL-framework/ippl-build-scripts). See the [documentation](https://github.com/IPPL-framework/ippl-build-scripts#readme) for more info on how to use the IPPL build script.
+### Optional requirements
+* FFTW
+* CuFFT
+
+## Compilation
+IPPL is a CMake Project and can be configured by passing options in CMake syntax:
+```
+cmake <src_dir> -D<option>=<value>
+```
+#### None of the options have to be set explicitly, all have a default.
+
+The relevant options of IPPL are
+- IPPL_PLATFORMS, can be one of `SERIAL, OPENMP, CUDA, "OPENMP;CUDA"`, default `SERIAL`
+- `KOKKOS_VERSION`, default `4.1.00`
+- `HEFFTE_VERSION`, default `MASTER`
+- `HEFFTE_VERSION`, can be default `MASTER`
+- `ENABLE_SOLVERS`, default `OFF`
+- `ENABLE_FFT`, default `OFF`
+  - If `ENABLE_FFT` is set, `Heffte_ENABLE_CUDA` will default to `ON` if `IPPL_PLATFORMS` contains `cuda`
+  - Otherwise, `Heffte_ENABLE_AVX2` is enabled. FFTW has to be enable explicitly.
+- `Heffte_ENABLE_FFTW`, default `OFF` 
```

Comment:
Does it download and install FFTW if the option is ON?

Reply:
It doesnt download fftw, FFTW is not cmake based afaik, I can arrange some setup if you want but it's hacly

Reply:
That's ok.. I don't think we need to do that. The user can/should install FFTW separately. So I guess if the flag is on but heffte can't find FFTW then heffte itself will print the error message right? (like FFTW_DIR not found or something like that). If that is the case we don't need to do anything and leave it as it is..

Reply:
Yes, the configuration of heffte fails if fftw isnt available